backend/database_test.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"github.com/go-redis/redismock/v8"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestConnect(t *testing.T) {
+	t.Run("should check that connection is already established and return error", func(t *testing.T) {
+		//given
+		client, _ := redismock.NewClientMock()
+		d := Database{address: "localhost:8081", password: "", connection: client, ctx: context.Background()}
+
+		//when
+		err := d.Connect()
+
+		//then
+		assert.Equal(t, nil, err)
+	})
+	t.Run("should return error when trying to Ping database", func(t *testing.T) { //todo:test refactor
+
+		//given
+		d := Database{address: "localhost:8081", password: "", ctx: context.Background()}
+
+		//when
+		err := d.Connect()
+
+		//then
+		assert.Error(t, err)
+		assert.NotEqual(t, nil, d.connection)
+	})
+}
+
+func TestInsertToDB(t *testing.T) {
+	t.Run("should return error when connection is not initialized", func(t *testing.T) {
+		//given
+		_, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(false)
+		database := Database{connection: nil}
+
+		//when
+		err := database.InsertToDB("key", "value")
+
+		//then
+		assert.Error(t, err, errors.New("INSERTTODB: connection not initialized"))
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("should return error when error occurred in adding value to database", func(t *testing.T) {
+		//given
+		client, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(true)
+		database := Database{connection: client}
+		key, value := "key", "value"
+		clientMock.ExpectSet(key, value, 0)
+
+		//when
+		err := database.InsertToDB(key, value)
+
+		//then
+		assert.Error(t, err)
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Run("should insert data correctly using the InsertToDB function - returns no error", func(t *testing.T) {
+
+		//given
+		client, clientMock := redismock.NewClientMock()
+		clientMock.ClearExpect()
+		clientMock.MatchExpectationsInOrder(true)
+		database := Database{connection: client}
+		key, value := "key", "value"
+		clientMock.ExpectSet(key, value, 0).SetVal(value)
+
+		//when
+		err := database.InsertToDB(key, value)
+
+		//then
+		assert.NoError(t, err)
+		//clientMock.ExpectGet(key).SetVal(value)
+		//val, err := database.GetFromDB(key)
+		//assert.NoError(t, err)
+		//assert.Equal(t, value, val)
+		if err := clientMock.ExpectationsWereMet(); err != nil {
+			t.Error(err)
+		}
+	})
+}
+
+func TestGetFromDB(t *testing.T) {
+	t.Run("should return error when connection is not initialized", func(t *testing.T) {
```

Comment:
You use `established` or `initialised` - try to stick to one naming for consistency